File: Frontend/Frontend/src/app/services/assignment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import {
  AssignmentCreateRequest,
  AssignmentResponse,
  AssignmentSubmissionRequest,
  AssignmentSubmissionResponse,
  GradeSubmissionRequest
} from '../dtos/assignment.dto';

@Injectable({
  providedIn: 'root'
})
export class AssignmentService {
  private readonly apiUrl = `${environment.apiUrl}/assignments`;

  constructor(private http: HttpClient) {}

  create(request: AssignmentCreateRequest): Observable<AssignmentResponse> {
    return this.http.post<AssignmentResponse>(this.apiUrl, request);
  }

  publish(assignmentId: number): Observable<AssignmentResponse> {
    return this.http.post<AssignmentResponse>(`${this.apiUrl}/${assignmentId}/publish`, {});
  }

  listForCourse(courseId: number): Observable<AssignmentResponse[]> {
    return this.http.get<AssignmentResponse[]>(`${this.apiUrl}/course/${courseId}`);
  }

  submit(assignmentId: number, request: AssignmentSubmissionRequest): Observable<AssignmentSubmissionResponse> {
    return this.http.post<AssignmentSubmissionResponse>(`${this.apiUrl}/${assignmentId}/submit`, request);
  }

  listSubmissions(assignmentId: number): Observable<AssignmentSubmissionResponse[]> {
    return this.http.get<AssignmentSubmissionResponse[]>(`${this.apiUrl}/${assignmentId}/submissions`);
  }

  grade(submissionId: number, request: GradeSubmissionRequest): Observable<AssignmentSubmissionResponse> {
    return this.http.post<AssignmentSubmissionResponse>(`${this.apiUrl}/submissions/${submissionId}/grade`, request);
  }

  listMySubmissions(courseId?: number): Observable<AssignmentSubmissionResponse[]> {
    let params = new HttpParams();
    if (courseId) params = params.set('courseId', courseId.toString());
    return this.http.get<AssignmentSubmissionResponse[]>(`${this.apiUrl}/submissions/me`, { params });
  }

  listPendingSubmissions(courseId: number): Observable<AssignmentSubmissionResponse[]> {
    return this.http.get<AssignmentSubmissionResponse[]>(`${this.apiUrl}/course/${courseId}/submissions/pending`);
  }
}

File: elearning-platform/elearning-platform/src/main/java/com/elearning/elearning_platform/assignment/service/AssignmentService.java
package com.elearning.elearning_platform.assignment.service;

import com.elearning.elearning_platform.assignment.dto.*;

import java.util.List;

public interface AssignmentService {
    AssignmentResponse create(Long actorId, String actorRole, AssignmentCreateRequest request);
    AssignmentResponse publish(Long actorId, String actorRole, Long assignmentId);
    List<AssignmentResponse> listForCourse(Long actorId, String actorRole, Long courseId);

    AssignmentSubmissionResponse submit(Long studentId, String actorRole, Long assignmentId, AssignmentSubmissionRequest request);
    List<AssignmentSubmissionResponse> listSubmissions(Long actorId, String actorRole, Long assignmentId);
    AssignmentSubmissionResponse grade(Long actorId, String actorRole, Long submissionId, GradeSubmissionRequest request);

    // Missing methods added
    List<AssignmentSubmissionResponse> listMySubmissions(Long studentId, Long courseId);
    List<AssignmentSubmissionResponse> listPendingSubmissions(Long actorId, String actorRole, Long courseId);
}